lib/presets.js

```diff
@@ -305,21 +305,59 @@ module.exports = {
      * });
      */
     cargo: (options, cwd, version, callback) => {
-      const cargoFiles = path.join(cwd, 'Cargo.*');
+      const cargoToml = path.join(cwd, 'Cargo.toml');
+      const cargoLock = path.join(cwd, 'Cargo.lock');
       const cleanedVersion = semver.clean(version);
 
       if (!cleanedVersion) {
         return callback(new Error(`Invalid version: ${version}`));
       }
 
-      replace({
-        files: cargoFiles,
-        from: /(version\s*=\s*)"(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)"/g,
-        to: '$1"' + cleanedVersion + '"'
-      }, (error) => {
+      fs.readFile(cargoToml, {
+        encoding: 'utf8'
+      }, (error, contents) => {
         if (error) {
+          if (error.code === 'ENOENT') {
+            error.message = `No such file or directory: ${cargoToml}`;
+          }
+
           return callback(error);
         }
+
+        // Capture first `name = "..."` occurrence immediately after `[package]`
+        const matches = contents.match(/\[package\](?:\n|.)*?name\s*=\s*(?:"|')(.+)(?:"|')/m);
+        if (_.isNull(matches)) {
+          return callback(
+            new Error(`Package name not found in ${cargoToml}`)
+          );
+        }
+        const packageName = matches[1];
+
+        // Update first `version = "..."` occurrence immediately after `[package]`
+        replace({
+          files: cargoToml,
+          from: /(\[package\](?:\n|.)*?version\s*=\s*)("|')(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)("|')/m,
+          to: '$1$2' + cleanedVersion + '$3'
+        }, (err) => {
+          if (err) {
+            return callback(err);
+          }
+        });
+
+        // Update first `version = "..."` occurrence immediately after `name = "${packageName}"`
```

Comment:
Should this whole chunk (and also the bit that extracts the packageName) be inside a `if (fs.existsSync(cargoLock))` ?

Reply:
Yes, it should. Nice catch!